walrus/methods/input/process_media.py

```diff
@@ -1646,38 +1662,36 @@ def process_one_image_file(self):
 
         """
 
-
-        result = self.read_raw_file()
-        if result is False: return False
+        # Read file if it does not come from a blob
+        if self.input.type != 'from_blob_path':
+            result = self.read_raw_file()
+            if result is False:
+                return False
 
         # Image() subclass
         self.new_image = Image(
             original_filename = self.input.original_filename)
+
+
         self.session.add(self.new_image)
         self.session.flush()
+        if self.input.type != 'from_blob_path':
+            self.new_image.url_signed_blob_path = settings.PROJECT_IMAGES_BASE_DIR + \
+                                                  str(self.project_id) + "/" + str(self.new_image.id)
+        else:
+            self.new_image.url_signed_blob_path = self.input.raw_data_blob_path
 
         self.try_to_commit()
 
         if self.project:
             self.project_id = self.project.id
 
-        ### Main
-
-        self.resize_raw_image()
-
-        self.check_metadata_and_auto_correct_instances(
-            imageio_read_image = self.raw_numpy_image,
-            image_metadata = self.input.image_metadata
-        )
+        # Save thumbnails if we are uploading blobs, otherwise skip
+        if self.input.type != 'from_blob_path':
```

Comment:
are we saving regening thumbnails in this context for future iteration?

Reply:
For now we are not regenerating anything as we don't have the context if the connection will allow writing or if its desirable to write in the external bucket provided in the connection. So for now my  reasoning was to skip it. But we can further discuss altternatives you would like to try

Reply:
That makes sense! 
Future, I like the idea of (optionally) reading the blob still, and then writing the thumb artifact somewhere